react/week3/todo-list-app/src/components/TodoList.js

```diff
@@ -0,0 +1,254 @@
+              ref={(value) => {
+                this.input = value;
+              }}
+              defaultValue={this.props.todo.description}
+              required
+            />{" "}
+            | {this.props.todo.deadline}
+            <input
+              type="checkbox"
+              checked={this.props.todo.checked}
+              onChange={() => {
+                this.props.handleCheckbox(this.props.index);
+              }}
+            />
+            <button
+              className="delete-button"
+              onClick={() => {
+                this.props.deleteTodo(this.props.index);
+              }}
+            >
+              Delete
+            </button>
+            <button type="submit">Update</button>
+          </form>
+        </li>
+      </FancyBorder>
+    );
+  };
+
+  renderTodoList = () => {
+    return (
+      <FancyBorder>
+        <li
+          style={{
+            textDecoration: this.props.todo.checked ? "line-through" : "none",
+          }}
+        >
+          <label>
+            {this.props.todo.description} | {this.props.todo.deadline}
+          </label>
+          <input
+            type="checkbox"
+            checked={!!this.props.todo.checked}
+            onChange={() => {
+              this.props.handleCheckbox(this.props.index);
+            }}
+          />
+          <button
+            className="delete-button"
+            onClick={() => {
+              this.props.deleteTodo(this.props.index);
+            }}
+          >
+            Delete
+          </button>
+          <button
+            onClick={(event) => {
+              event.stopPropagation();
+              this.toggleState();
+            }}
+          >
+            Edit
+          </button>
+        </li>
+      </FancyBorder>
+    );
+  };
+  render() {
+    return (
+      <div>
+        {this.state.isEditing ? this.renderEditMode() : this.renderTodoList()}
+      </div>
+    );
+  }
+}
+
+function fetchTodo() {
+  const URL = `https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw`;
+  return fetch(URL).then((response) => response.json());
+}
+
+class TodoList extends React.Component {
+  state = {
+    description: "",
+    deadline: "",
+    checked: false,
+    todoList: [],
+  };
+
+  componentDidMount() {
+    this.fetchTodos();
+  }
+
+  fetchTodos = () => {
+    fetchTodo().then((todo) => {
+      this.setState({ todoList: todo });
+    });
+  };
+
+  handleSubmit = (event) => {
+    this.setState({
+      todoList: [
+        ...this.state.todoList,
+        {
+          description: this.state.description,
+          deadline: this.state.deadline,
+          checked: false,
+        },
+      ],
+    });
+    event.preventDefault();
+  };
+
+  handleDescriptionChange = (event) => {
+    this.setState({ description: event.target.value });
+  };
+
+  handleDeadlineChange = (event) => {
+    let currentDate = new Date();
+    let selectedDate = new Date(event.target.value);
```

Comment:
Use `const` instead of `let` if you know that the value of the variable won't change.

Reply:
implemented the suggested correction.